code/__HELPERS/dna.dm

```diff
@@ -11,3 +11,13 @@
 #define GET_MUTATION_SYNCHRONIZER(A) ((A.synchronizer_coeff < 0) ? 1 : A.synchronizer_coeff)
 #define GET_MUTATION_POWER(A) ((A.power_coeff < 0) ? 1 : A.power_coeff)
 #define GET_MUTATION_ENERGY(A) ((A.energy_coeff < 0) ? 1 : A.energy_coeff)
+
+#define GET_DNA_BLOCK(input, blocknum, len_by_block) copytext(input, len_by_block[blocknum], LAZYACCESS(len_by_block, blocknum+1))
+
+#define GET_UI_BLOCK(input, blocknum) GET_DNA_BLOCK(input, blocknum, GLOB.total_ui_len_by_block)
+#define GET_UF_BLOCK(input, blocknum) GET_DNA_BLOCK(input, blocknum, GLOB.total_uf_len_by_block)
+
+///Getter macro used to get the length of a identity block
+#define GET_UI_BLOCK_LEN(blocknum) (GLOB.identity_block_lengths["[blocknum]"] || DNA_BLOCK_SIZE)
+///Ditto, but for a feature.
+#define GET_UF_BLOCK_LEN(blocknum) (GLOB.features_block_lengths["[blocknum]"] || DNA_BLOCK_SIZE)
```

Comment:
Why do these need to be macros? It's a lot of behavior to put in one place, and again, people are bad at reading these.

```
why is this meeting not a phone call
why is this phone call not an email
why is this email not a text
why is this text not a proc
```

Reply:
You're right with the first three ones, but the block length getters are fine as is; all they do is access the assoc value of a key global list or return the default size.

Reply:
My beef is less with the length getters and more GET_DNA_BLOCK, since it's a lot of context about what's actually happening here put in a one liner. I don't understand why it should be a macro instead of a proc. Is it to make it easier to use with `GLOB.total_uf_len_by_block` and such?